src/components/Tools/ProgressTracker.vue

```diff
@@ -0,0 +1,269 @@
+<template>
+  <div class="progress">
+    <div class="progress-position">
+      <div class="progress-bar-container">
+        <div class="progress-bar-overflow">
+          <div
+            class="progress-bar"
+            :style="{ transform: `rotate(${45 + 180 * sumRequirements()}deg)` }"
+          ></div>
+        </div>
+        <img class="progress-bar-caption" :src="getImage" />
+      </div>
+      <div class="progress-status">
+        <span class="progress-status-text">{{ getProgressString }}</span>
+      </div>
+    </div>
+    <div class="progress-text">
+      <span class="progress-text-style"> You've completed: </span>
+      <span class="progress-text-style" v-if="hasCollege">
+        {{ collegeRequirementCount.finished }} / {{ collegeRequirementCount.needed }} College
+        Requirements
+      </span>
+      <span class="progress-text-style" v-if="hasMajor">
+        {{ majorRequirementCount.finished }} / {{ majorRequirementCount.needed }} Major
+        Requirements</span
+      >
+      <span class="progress-text-style" v-if="hasMinor">
+        {{ minorRequirementCount.finished }} / {{ minorRequirementCount.needed }} Minor
+        Requirements</span
+      >
+      <span class="progress-text-style" v-if="hasGrad">
+        {{ gradRequirementCount.finished }} / {{ gradRequirementCount.needed }} Grad
+        Requirements</span
+      >
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import store from '@/store';
+import confetti from '@/assets/images/progress_tracker/confetti.svg';
+import fire from '@/assets/images/progress_tracker/fire.svg';
+import flex from '@/assets/images/progress_tracker/flex.svg';
+import hands from '@/assets/images/progress_tracker/hands_raised.svg';
+import star from '@/assets/images/progress_tracker/star.svg';
+
+export default defineComponent({
+  data() {
+    return {
+      progressImg: {
+        Confetti: confetti,
+        Fire: fire,
+        Flex: flex,
+        Hands: hands,
+        Star: star,
+      },
```

Comment:
These are constants right? I think this can be removed if you make the change I suggested below.